cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -692,7 +692,12 @@ pub mod pallet {
 				vfp.relay_parent_number,
 			));
 
-			Ok(PostDispatchInfo { actual_weight: Some(total_weight), pays_fee: Pays::No })
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
+				total_weight,
```

Comment:
I assume we don't know an upper bound for this value pre-dispatch because it is calculated ad-hoc?

Reply:
Yes but it's once per block and we register the weight, meaning it will be correctly accounted for in the following transactions.

If we go with the worst case weight here before dispatching, we might greatly underestimate the block space we have since those stuctures in `ParachainInherentData` can grow a lot as far as I'm aware. This model works best IMO.